### PR TITLE
Lazy load voice class & Add database support for aoi.mongo

### PR DESCRIPTION
**Pull request by Azusfin**
## Type
- [ ] Bug Fix
- [ ] Functions: \<Function Name>
- [ ] Callbacks: \<Callback Name>
- [ ] Handlers: \<Handler Type>
- [x] Others:
  - Lazy load voice class
  - Add database support for aoi.mongo

Dependencies (Third Party Modules) needed: <Name | Github Link> (Maximum: 20MB~ size)

Want a credit? Discord tag or other social media link: <DiscordTag | OtherSocialMediaLink>

Referenced Issue: #NaN (Answer for an issue, if any)

## Description
- Lazy load voice class so it will only return error when constructed
- Add database support for aoi.mongo
  How to use:
    - Install `mongodb`
    - Install `aoi.mongo`
    - Example setup:
      where `prefix` is the bot prefix and `mongo cluster url` is the mongodb cluster url like
      
      `mongodb+srv://<username>:<password>@<???>.<???>.mongodb.net/?w=majority&retryWrites=true`
      ```js
      const aoijs = require("aoi.js")
      const aoimongo = require("aoi.mongo")
      
      const bot = new aoijs.Bot({
        token: process.env.TOKEN,
        prefix: "prefix",
        intents: "all",
        database: {
          db: aoimongo,
          type: "aoi.mongo",
          path: "mongo cluster url"
        }
      })
      
      // ...
      ```

### CODE REVIEW

File: src/functions/Funcs/info/dbPing.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 module.exports = async d => {
     const {code} = d.util.openFunc(d);
+    const ping = d.client.db.roundTrip ? await d.client.db.roundTrip() : d.client.db.ping
```
Comment (Leref) [inline src/functions/Funcs/info/dbPing.js:3]: A bit unethical.

File: src/functions/Funcs/misc/serverLeaderBoard.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
-const {AoijsAPI, DbdTsDb, Promisify, CustomDb} = require("../../../classes/Database.js");
+const {AoijsAPI, DbdTsDb, AoiMongoDb, Promisify, CustomDb} = require("../../../classes/Database.js");
 
 module.exports = async d => {
     const Data = d.util.openFunc(d);
     if (Data.err) return d.error(Data.err)
 
-    const [variable, type = 'asc', custom = `{top}) {username} : {value}`, list = 10, page = 1, table = d.client.db.tables[0]] = Data.inside.splits;
```
Comment (Leref) [inline src/functions/Funcs/misc/serverLeaderBoard.js:7]: This shouldn't be changed.

**Review by Leref — CHANGES_REQUESTED** [review]

**Comment by USERSATOSHI** [comment]
can you also complete the var related functions

File: src/functions/Funcs/info/dbPing.js (hunk 1 above)
Comment (Azusfin) [reply src/functions/Funcs/info/dbPing.js]: Is this okay? 4d2b95f6e468012938e22eadcc13b2da7a1af3cd

**Comment by Azusfin** [comment]
> can you also complete the var related functions

what kind?

**Review by Leref — APPROVED** [review]

**Comment by USERSATOSHI** [comment]
like if u check leaderboard functions
u can see there is a instanceof checker for each class type

**Comment by Azusfin** [comment]
> like if u check leaderboard functions u can see there is a instanceof checker for each class type

src/functions/Funcs/misc/rawLeaderboard.js src/functions/Funcs/misc/globalUserLeaderBoard.js src/functions/Funcs/misc/userLeaderBoard.js src/functions/Funcs/misc/serverLeaderBoard.js

do i miss something?

**Comment by USERSATOSHI** [comment]
btw prob should make it {name} instead of {username} in serverlb cause {username} wont make sense tbh

**Comment by Azusfin** [comment]
> btw prob should make it {name} instead of {username} in serverlb cause {username} wont make sense tbh

it was, but leref complaining that `this shouldn't be changed.` with dot

**Comment by USERSATOSHI** [comment]
> > btw prob should make it {name} instead of {username} in serverlb cause {username} wont make sense tbh
> 
> it was, but leref complaining that `this shouldn't be changed.` with dot

a ok then i cant complain

**Comment by USERSATOSHI** [comment]
lgtm

**Comment by DevMike123** [comment]
How to retrieve data?

**Comment by Azusfin** [comment]
> How to retrieve data from the database?

$getVar $getGlobalUserVar $getUserVar $getServerVar ...

**Comment by DevMike123** [comment]
Mongo Database!

**Comment by USERSATOSHI** [comment]
its same

**Comment by DevMike123** [comment]
Thanks

**Comment by DevMike123** [comment]
Do we need to give the name of database in mongo cluster uri.

**Comment by Azusfin** [comment]
> Do we need to give the name of database in mongo cluster uri.

no need, it's default to "aoijs"
but you can change it in extraOptions property

```
database: {
  ...,
  extraOptions: {
    databaseName: "dbname"
  }
}```

**Comment by DevMike123** [comment]
> > Do we need to give the name of database in mongo cluster uri.
> 
> no need, it's default to "aoijs"
> but you can change it in extraOptions property
> 
> ```
> database: {
>   ...,
>   extraOptions: {
>     databaseName: "dbname"
>   }
> }
> ```

1. Do we need to write collection name?

2. Is aoi.mongo suitable with aoi.js v4?

**Comment by Azusfin** [comment]
1. collection name is the table name...
2. no, aoijs v4 doesn't support aoimongo and it's also deprecated anyway

**Comment by DevMike123** [comment]
> 1. collection name is the table name...
> 2. no, aoijs v4 doesn't support aoimongo and it's also deprecated anyway

Are mongoose and MongoDB suitable with aoi.js v4?

**Comment by Azusfin** [comment]
v4 will need to use dbdjs.mongo for mongodb.
why are you still using v4

**Comment by DevMike123** [comment]
> v4 will need to use dbdjs.mongo for mongodb.
> why are you still using v4

Does dbdjs.mongo work with $getServerVar?

**Comment by Azusfin** [comment]
...

**Comment by Azusfin** [comment]
please stop with the useless questions.

**Comment by USERSATOSHI** [comment]
> 

if aoijs supports it then yes else no

**Pull request merged**